seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -495,10 +495,11 @@ public PassiveCompletableFuture<Void> savePoint(long jobId) {
     public PassiveCompletableFuture<JobResult> waitForJobComplete(long jobId) {
         JobMaster runningJobMaster = runningJobMasterMap.get(jobId);
         if (runningJobMaster == null) {
-            JobStatus jobStatus = jobHistoryService.getJobDetailState(jobId).getJobStatus();
+            JobHistoryService.JobState jobState = jobHistoryService.getJobDetailState(jobId);
             CompletableFuture<JobResult> future = new CompletableFuture<>();
-            // TODO support history service record job execute error
-            future.complete(new JobResult(jobStatus, null));
+            if (jobState == null) future.complete(new JobResult(JobStatus.FAILED, null));
```

Comment:
How about changing null (after JobStatus.FAILED) to alert that the job failed？

Reply:
Here I think we should add a new state to notify the user that the Job is lost, instead of returning FAILED, but the impact of adding the state modification scope is large, so I think we can optimize it later!

Reply:
Agree with you.